src/main/java/org/testng/GuiceHelper.java

```diff
@@ -33,15 +40,38 @@ Injector getInjector(IClass iClass) {
     List<Module> moduleInstances =
         Lists.newArrayList(getModules(guice, parentInjector, iClass.getRealClass()));
 
-    // Reuse the previous injector, if any
+    if (isStringNotEmpty(context.getSuite().getParentModule())) {
+      Class<Module> parentModule = (Class<Module>) ClassHelper.forName(context.getSuite().getParentModule());
```

Comment:
In fact, it should be `Class<?> parentModule` here and you'll be able to remove the suppress warning.
At this moment, we don't know what kind of class the user is providing.

Reply:
yeah, still need to cast to a `Module` at some point, but can avoid casting to `Class<Module>` and thereby avoid the unchecked suppression